kivy/uix/textinput.py

```diff
@@ -566,6 +566,9 @@ def __init__(self, **kwargs):
         fbind('font_family', refresh_line_options)
         fbind('base_direction', refresh_line_options)
         fbind('text_language', refresh_line_options)
+        fbind('foreground_color', refresh_line_options)
+        fbind('disabled_foreground_color', refresh_line_options)
+        fbind('hint_text_color', refresh_line_options)
```

Comment:
I recommend binding to `update_text_options`, binding to `refresh_line_options` will move the cursor to the end of the text.

So you can move this code below, to the line below [line 588](https://github.com/kivy/kivy/pull/7997/files#diff-67d848ba06b2f91032fd6dc59a8956275118595997f9737872a39fb5983183d2R588)

```python
fbind('foreground_color', update_text_options)
fbind('disabled_foreground_color', update_text_options)
fbind('hint_text_color', update_text_options)
```